server/src/main/java/org/elasticsearch/index/mapper/blockloader/docvalues/MvMaxBytesRefsFromOrdsBlockLoader.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper.blockloader.docvalues;
+
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+
+/**
+ * Loads the MAX {@code keyword} in each doc.
+ */
+public class MvMaxBytesRefsFromOrdsBlockLoader extends AbstractBytesRefsFromOrdsBlockLoader {
+    private final String fieldName;
+
+    public MvMaxBytesRefsFromOrdsBlockLoader(String fieldName) {
+        super(fieldName);
+        this.fieldName = fieldName;
+    }
+
+    @Override
+    protected AllReader singletonReader(SortedDocValues docValues) {
+        return new Singleton(docValues);
+    }
+
+    @Override
+    protected AllReader sortedSetReader(SortedSetDocValues docValues) {
+        return new MvMaxSortedSet(docValues);
+    }
+
+    @Override
+    public String toString() {
+        return "MvMaxBytesRefsFromOrds[" + fieldName + "]";
+    }
+
+    private static class MvMaxSortedSet extends BlockDocValuesReader {
+        private final SortedSetDocValues ordinals;
+
+        MvMaxSortedSet(SortedSetDocValues ordinals) {
+            this.ordinals = ordinals;
+        }
+
+        @Override
+        public Block read(BlockFactory factory, Docs docs, int offset, boolean nullsFiltered) throws IOException {
+            if (docs.count() - offset == 1) {
+                return readSingleDoc(factory, docs.get(offset));
+            }
+            try (var builder = factory.sortedSetOrdinalsBuilder(ordinals, docs.count() - offset)) {
```

Comment:
We should use singletonOrdinalsBuilder since each row should have at most one value. However, we are not using it because it requires SortedDocValues. This is not an issue here, but if this pattern recurs, we might consider relaxing singletonOrdinalsBuilder to accept an interface that supports looking up terms by ordinal.